bookstore-rust/api/src/routes.rs

```diff
@@ -0,0 +1,69 @@
+use actix_web::{delete, get, HttpResponse, post, Result, web};
+use actix_web::error::{ErrorInternalServerError, ErrorNotFound};
+use uuid::Uuid;
+
+use bookstore_core::Mutation;
+use bookstore_core::Query;
+use entity::book;
+
+use crate::dtos::{Book, CreateBook, CreateBookResponse};
+use crate::server::AppState;
+
+#[get("/books/{id}")]
+pub async fn get_book_by_id(data: web::Data<AppState>, path_id: web::Path<Uuid>) -> Result<web::Json<Book>> {
+    let conn = &data.conn;
```

Comment:
Better don't use the db in the controller. In the other examples, we always use a service in between.

Reply:
I know ... this is basically taken from multiple examples as a "actix/rust" way and best practice there ... I tried to stay close to the recommended patterns.

Shu Ha Ri .... I am currently a lot in "Shu" :)

Reply:
It would be interesting to see the impact with Depedency Injection because we also have this capability in the other implementations.

Reply:
Yes, good point ... I will try to find out how DI is done in Rust